subsys/bluetooth/host/classic/hfp_ag.c

```diff
@@ -0,0 +1,2092 @@
+	va_end(vargs);
+
+	if (ret < 0) {
+		LOG_ERR("Unable to format variable arguments");
+		net_buf_unref(buf);
+		bt_ag_tx_free(tx);
+		return ret;
+	}
+
+	net_buf_add(buf, ret);
+
+	key = irq_lock();
+	sys_slist_append(&ag->tx_pending, &tx->node);
+	irq_unlock(key);
+
+	ret = hfp_ag_send(ag, tx);
+
+	if (ret < 0) {
+		LOG_ERR("Rfcomm send error :(%d)", ret);
+		key = irq_lock();
+		sys_slist_find_and_remove(&ag->tx_pending, &tx->node);
+		irq_unlock(key);
+		net_buf_unref(buf);
+		bt_ag_tx_free(tx);
+		return ret;
+	}
+
+	return 0;
+}
+
+static void skip_space(struct net_buf *buf)
+{
+	while ((buf->len > 0) && (buf->data[0] == ' ')) {
+		(void)net_buf_pull(buf, 1);
+	}
+}
+
+static int get_number(struct net_buf *buf, uint32_t *number)
+{
+	int err = -EINVAL;
+
+	*number = 0;
+
+	skip_space(buf);
+	while ((buf->len > 0)) {
+		if ((buf->data[0] >= '0') && (buf->data[0] <= '9')) {
+			*number = *number * 10 + buf->data[0] - '0';
+			(void)net_buf_pull(buf, 1);
+			err = 0;
+		} else {
+			break;
+		}
+	}
+	skip_space(buf);
+
+	return err;
+}
+
+static bool is_char(struct net_buf *buf, uint8_t c)
+{
+	bool found = false;
+
+	skip_space(buf);
+	if ((buf->len > 0)) {
+		if (buf->data[0] == c) {
+			(void)net_buf_pull(buf, 1);
+			found = true;
+		}
+	}
+	skip_space(buf);
+
+	return found;
+}
+
+static int bt_hfp_ag_brsf_handler(struct bt_hfp_ag *ag, struct net_buf *buf, enum at_cme *err)
+{
+	uint32_t number;
+	int error;
+	bool ret;
+
+	*err = CME_ERROR_AG_FAILURE;
+
+	ret = is_char(buf, '=');
+	if (ret != true) {
+		return -EINVAL;
+	}
+
+	error = get_number(buf, &number);
+
+	if (error != 0) {
+		return error;
+	}
+
+	ret = is_char(buf, '\r');
+	if (ret != true) {
+		return -EINVAL;
+	}
+
+	ag->hf_features = number;
+
+	return hfp_ag_send_data(ag, NULL, NULL, "\r\n+BRSF:%d\r\n", ag->ag_features);
+}
+
+static int bt_hfp_ag_bac_handler(struct bt_hfp_ag *ag, struct net_buf *buf, enum at_cme *err)
+{
+	uint32_t number;
+	uint32_t codec_ids = 0U;
+	int error;
+	bool ret;
+
+	*err = CME_ERROR_AG_FAILURE;
+
+	ret = is_char(buf, '=');
+	if (ret != true) {
+		return -EINVAL;
+	}
+
+	if (!(ag->hf_features & BT_HFP_HF_FEATURE_CODEC_NEG) ||
+	    !(ag->ag_features & BT_HFP_AG_FEATURE_CODEC_NEG)) {
+		return -EOPNOTSUPP;
+	}
+
+	while (error == 0) {
+		error = get_number(buf, &number);
+
+		if (error != 0) {
+			return error;
+		}
+
+		ret = is_char(buf, ',');
+		if (ret != true) {
+			if (true != is_char(buf, '\r')) {
+				return -EINVAL;
+			}
+		}
+
+		if (number < (sizeof(codec_ids) * 8)) {
+			codec_ids |= BIT(number);
+		}
+	}
+
+	ag->hf_codec_ids = codec_ids;
+
+	if (bt_ag && bt_ag->codec) {
+		bt_ag->codec(ag, ag->hf_codec_ids);
+	}
+
+	return 0;
+}
+
+static int bt_hfp_ag_cind_handler(struct bt_hfp_ag *ag, struct net_buf *buf, enum at_cme *err)
+{
+	int error;
+	bool inquiry_status = true;
+	bool ret;
+
+	*err = CME_ERROR_AG_FAILURE;
+
+	ret = is_char(buf, '=');
+	if (ret == true) {
```

Comment:
```c
	if (is_char(buf, '=')) {
		inquiry_status = false;
	}
```

There are several more places like this which I won't comment on separately - just fix all of them please :)